### PR TITLE
Updating test/integration to use testapi.Version everywhere

### PR DESCRIPTION
**Pull request by nikhiljindal**
For https://github.com/GoogleCloudPlatform/kubernetes/issues/6584
Updating test/integration to use testapi.Version everywhere.

Verified that `grep -r "v1beta1" test/integration/` does not return anything after this change.

### CODE REVIEW

**Comment by bprashanth** [comment]
LGTM, just feel weird about testapi.Version using the env variable but unable to make a better suggestion right now (@lavalamp mentioned there might already be a flag that does this? otherwise breaking out the flag package might be overkill)

**Comment by lavalamp** [comment]
Yeah-- I was misremembering the env var. hack/test-go.sh and test-integration already set this env var.

File: pkg/api/testapi/testapi.go [hunk 1]
```diff
@@ -76,14 +77,21 @@ func SelfLink(resource, name string) string {
...
 
-// Returns the appropriate path for the given resource, namespace and name.
+// Returns the appropriate path for the given prefix (watch, proxy, redirect, etc), resource, namespace and name.
 // For ex, this is of the form:
-// /api/v1beta1/pods/pod0 for v1beta1 and
-// /api/v1beta3/namespaces/foo/pods/pod0 for v1beta3.
-func ResourcePath(resource, namespace, name string) string {
+// /api/v1beta1/watch/pods/pod0 for v1beta1 and
+// /api/v1beta3/watch/namespaces/foo/pods/pod0 for v1beta3.
+func ResourcePathWithPrefix(proxy, resource, namespace, name string) string {
```
Comment (lavalamp) [inline pkg/api/testapi/testapi.go]: please write a unit test for this, and also rename proxy to prefix ;)

Comment (lavalamp) [inline pkg/api/testapi/testapi.go]: Might also consider shortening these names-- they make those table driven tests pretty hard to read :(

Comment (nikhiljindal) [inline pkg/api/testapi/testapi.go]: Done.

**Comment by nikhiljindal** [comment]
Updated the code as per comments. Ready for another look.

**Comment by nikhiljindal** [comment]
Travis is green now :)

**Comment by nikhiljindal** [comment]
Can I haz mergez?

**Pull request merged**